### ISSUE TITLE
Default options given via GET are not proeprly kept when navigating since Vue

### ISSUE THREAD

**Issue by proton-ab** [thelounge/thelounge#3567]
- Node version: v10.17.0
- Browser version: Any
- Device, operating system: Any
- The Lounge version: 2a84d8239b17dd9f384ac777278aaafe0d46588a

---

Since Vue merge, supplying default set of options via GET parameter (for example "?nick=&channels=support&username=") is partially broken as navigating to Options or Help page and then back to Connect page resets the options as the URL is changed and GET parameters are lost in the process. This generally affects public mode installations.

**Comment by xPaw** [comment]
Before Vue, url parameters have overriden the default parameters, and it stayed that way. In vue, it uses url parameters to navigate to `connect` with said parameters (pressing back in the browser should give you that).

@richrd Do we want to support this?

**Comment by richrd** [comment]
Seems like a bit of an edge case to support although I'm really against it. Not sure what the best way to fix it would be.

**Comment by xPaw** [comment]
Yeah, I'm not really keen on overriding the defaults that come from server with the ones provided in the url (that's what it did before).

**Comment by proton-ab** [comment]
Might be an edge case but pretty useful when in public mode and used as IRC widget (for example it could be embedded via iframe)

Let's look at it differently too - ignore url parameter thing, have user type some username and channel, then have him navigate to Settings and back to Connect page - everything typed is now removed.

**Comment by richrd** [comment]
Now you mentioned that use case I'm starting to feel this is more of an issue. You're right that the window is re-rendered and that's why the fields reset. Might take a look at this when I have time.

**Comment by xPaw** [comment]
It reset the fields before Vue too, that's not a new 'issue'.

**Comment by richrd** [comment]
Oh, right. I was unaware of that. I think it could still be a useful feature not to reset them though.

**Comment by xPaw** [comment]
I also noticed that when clicking Connect, most fields reset/redraw to their default values before the window disappears.

We probably need to make all the inputs use `v-model` to fix this weirdness.
